src/main/java/com/example/demo/security/configuration/ApplicationSecurity.java

```diff
@@ -52,11 +52,12 @@ protected void configure(HttpSecurity http) throws Exception {
 
         // authorises any request via /auth/login to enter without authentication
         http.authorizeRequests()
-                .antMatchers("/auth/login").permitAll()
+                .antMatchers("/auth/login", "/h2-console/**", "/public/**").permitAll()
                 .anyRequest().authenticated();
 
         // This exception handling code ensures that the server will return HTTP status 401 (Unauthorized)
         // if any error occurs during authentication process
+        // todo: this is not what I want. It is causing nonsense urls to return 401 unauthorised errors. Which is not useful
```

Comment:
Did you follow a tutorial for the exception handling is that why it's behaving this way?

Reply:
yeah..
Without this bit of code, we get a 403 response for trying to access a private endpoint without a token

Reply:
Let me know if you need me to look at anything!